pints/_mcmc/_smc.py

```diff
@@ -0,0 +1,266 @@
+#
+# Uses the Python `pymc3` (and `theano`) module to run SMC routine sampling.
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+import pints
+import numpy as np
+import theano
+import theano.tensor as tt
+from theano.compile.ops import as_op
+import pymc3 as pm
+# from pymc3.step_methods import smc
+
+
+# Set some flags for theano
+THEANO_FLAGS = 'floatX=float32,optimizer=fast_compile'
+theano.config.floatX = 'float64'
+theano.config.compute_test_value = 'raise'
+theano.config.exception_verbosity = 'high'
+
+
+#TODO: issue due to @as_op cannot be done inside class
+global evaluate, times
```

Comment:
Will this work for repeated runs etc.? Usually a bad idea!

Reply:
Yes! This is a really bad idea indeed. But I tried putting the `@as_op` on line 244 inside the class, it does not like it. If we find a better way of dealing the `@as_op` from `theano`, then we should definitely remove this.